blog.py

```diff
@@ -332,6 +332,17 @@ def delete(self, posts: Union[str, List[str]]):
         sql = f'DELETE FROM posts WHERE path in ({qmark})'
         self._conn.execute(sql, posts)
 
+    def trash(self, posts: Union[str, List[str]]):
+        if isinstance(posts, str):
+            posts = [posts]
+        if not os.path.isdir(os.path.join(BASE_DIR, 'trash')):
+            os.mkdir(os.path.join(BASE_DIR, 'trash'))
+        for post in posts:
+            shutil.move(post, os.path.join(BASE_DIR, 'trash'))
+            qmark = ', '.join(['?'] * len(posts))
+            sql = f'DELETE FROM posts WHERE path in ({qmark})'
+            self._conn.execute(sql, posts) 
```

Comment:
Wrong indention. You have have to delete the posts once. Why do you place the code block in a loop?